webui/react/src/pages/TrialDetails.tsx

```diff
@@ -15,23 +15,68 @@ import usePolling from 'hooks/usePolling';
 import useRestApi from 'hooks/useRestApi';
 import TrialActions, { Action as TrialAction } from 'pages/TrialDetails/TrialActions';
 import TrialInfoBox from 'pages/TrialDetails/TrialInfoBox';
+import { routeAll } from 'routes';
+import { forkExperiment } from 'services/api';
 import { getExperimentDetails, getTrialDetails, isNotFound } from 'services/api';
 import { TrialDetailsParams } from 'services/types';
-import { ExperimentDetails, TrialDetails } from 'types';
+import { ExperimentDetails, RawJson, TrialDetails } from 'types';
 import { clone } from 'utils/data';
 import { trialHParamsToExperimentHParams } from 'utils/types';
 
 interface Params {
   trialId: string;
 }
 
+const getTrialLength = (experimentConfig: RawJson): [string, number] => {
+  const maxLength = experimentConfig.searcher.max_length;
+  if (!maxLength) {
+    // provide backward compat for max_steps
+    let maxSteps = experimentConfig.searcher.max_steps;
+    maxSteps = parseInt(maxSteps) || 100;
+    const batchesPerStep = experimentConfig.batches_per_step || 100;
+    return [ 'batches', maxSteps * batchesPerStep ];
+  }
+  return Object.entries(maxLength)[0] as [string, number];
+};
+
+const setTrialLength = (experimentConfig: RawJson, length: number): void => {
+  const lengthType = getTrialLength(experimentConfig)[0];
+  experimentConfig.searcher.max_length = { [lengthType]: length } ;
+  // in case we are dealing with an old config
+  delete experimentConfig.searcher.max_steps;
+};
+
+const trialContinueConfig =
+(experimentConfig: RawJson, trialHparams: Record<string, string>, trialId: number): RawJson => {
+  const newSearcher = {
+    max_length: experimentConfig.searcher.max_length,
+    metric: experimentConfig.searcher.metric,
+    name: experimentConfig.searcher.name,
+    smaller_is_better: experimentConfig.searcher.smaller_is_better,
+    source_trial_id: trialId,
+  };
+  const newHyperparameters = trialHParamsToExperimentHParams(trialHparams);
+
+  return {
+    ...experimentConfig,
+    description: experimentConfig.description,
+    hyperparameters: newHyperparameters,
+    searcher: newSearcher,
+  };
```

Comment:
**nit:** thinking it can just be the entire object defined and returned at the same time instead of declaring the smaller chunks and assigning them afterwards.

Reply:
agreed it used to be more involved here. but it seems small enough now.